test/parallel/test-runner-mocking.js

```diff
@@ -318,6 +318,68 @@ test('spy functions can be bound', (t) => {
   assert.strictEqual(sum.mock.restore(), undefined);
   assert.strictEqual(sum.bind(0)(2, 11), 13);
 });
+test('spies on async class functions', async (t) => {
+  class Runner {
+    async someTask(msg) {
+      return Promise.resolve(msg);
+    }
+
+    async method(msg) {
+      await this.someTask(msg);
+      return msg;
+    }
+  }
+  const msg = 'ok';
+  const obj = new Runner();
+  assert.strictEqual(await obj.method(msg), msg);
+
+  t.mock.method(obj, obj.someTask.name);
+  assert.strictEqual(obj.someTask.mock.calls.length, 0);
+
+  assert.strictEqual(await obj.method(msg), msg);
+
+  const call = obj.someTask.mock.calls[0];
+
+  assert.deepStrictEqual(call.arguments, [msg]);
+  assert.strictEqual(await call.result, msg);
+  assert.strictEqual(call.target, undefined);
+  assert.strictEqual(call.this, obj);
+
+  assert.strictEqual(obj.someTask.mock.restore(), undefined);
+  assert.strictEqual(await obj.method(msg), msg);
+  assert.strictEqual(obj.someTask.mock, undefined);
```

Comment:
Can you assert that the method is only defined on the prototype. We should not be restoring to the instance itself in this scenario.

Reply:
I think I didn't get it. In which part did you mean?

Reply:
You have `obj` which is an instance of `Runner`. The mocked method, `someTask()` comes from `Runner`. So, after calling `restore()`, we should verify that `Runner` has the original method, and `obj` should not have the method. Basically, we are checking that `restore()` gets us back to the original state before any mocking happened.

Reply:
Perfect, I'd just add this assertion, right?

```js
assert.strictEqual(Runner.prototype.someTask.mock, undefined);
```

Reply:
probably more explicitly, something like `assert.notOk('mock' in Runner.prototype.someTask)`, since absence is more correct than `undefined`?

Reply:
just added them to both static and instance objects